lib/licensed/source/cabal.rb

```diff
@@ -161,19 +165,22 @@ def cabal_file_dependencies
           content.scan(cabal_file_regex).each do |match|
             # match[1] is a string of "," separated dependencies
             dependencies = match[1].split(",").map(&:strip)
-            dependencies.each do |dep|
-              # the dependency might have a version specifier.
-              # remove it so we can get the full id specifier for each package
-              id = cabal_package_id(dep.split(/\s/)[0])
-              packages.add(id) if id
-            end
+
+            # the dependency might have a version specifier.
+            # remove it so we can get the full id specifier for each package
+            targets.merge(dependencies.map { |dep| dep.split(/\s/)[0] })
```

Comment:
What do you think about evaluating `strip` and `split` when initially `map`ing the initial `dependencies` list?

```ruby
dependencies = match[1].split(",").map { |dep| dep.strip.split(/\s/)[0] }
targets.merge(dependencies)
```

Reply:
sure, though I'm curious whether thats for stylistic taste or whether there's something deeper I'm missing?

updated at https://github.com/github/licensed/pull/75/commits/2c5483faecb1ccd2d5430fa3e41b8fa86e7824f2

Reply:
@jonabc ah sorry I didn't specify, this suggestion was intended to make traversing over the list of dependencies via a single pass rather than two `map`s.